src/ddmd/link.d

```diff
@@ -1039,4 +983,384 @@ version (Windows)
         const(char)* liblegacy = FileName.replaceName(libcmt, "legacy_stdio_definitions.lib");
         return FileName.exists(liblegacy) == 1;
     }
+
+    struct VSOptions
+    {
+        // evaluated once at startup
+        const(char)* WindowsSdkDir;
+        const(char)* WindowsSdkVersion;
+        const(char)* UCRTSdkDir;
+        const(char)* UCRTVersion;
+        const(char)* VSInstallDir;
+        const(char)* VisualStudioVersion;
+        const(char)* VCInstallDir;
+        const(char)* VCToolsInstallDir; // used by VS 2017
+
+        void initialize()
+        {
+            detectWindowsSDKDir();
+            detectUCRT();
+            detectVSInstallDir();
+            detectVCInstallDir();
+            detectVCToolsInstallDir();
+        }
+
+        const(char)* linkOptions(bool x64)
+        {
+            OutBuffer cmdbuf;
+            if (auto p = getVCDir(VCDir.Lib, x64))
+            {
+                cmdbuf.writestring(" /LIBPATH:\"");
+                cmdbuf.writestring(p);
+                cmdbuf.writeByte('\"');
+            }
+            if (VisualStudioVersion && strcmp(VisualStudioVersion, "14") >= 0)
+            {
+                if (auto p = getUCRTLibPath(x64))
+                {
+                    cmdbuf.writestring(" /LIBPATH:\"");
+                    cmdbuf.writestring(p);
+                    cmdbuf.writeByte('\"');
+                }
+                cmdbuf.writestring(" legacy_stdio_definitions.lib");
+            }
+            const(char)* windowssdkdir = getenv("WindowsSdkDir");
+            if (auto p = getSDKLibPath(x64))
+            {
+                cmdbuf.writestring(" /LIBPATH:\"");
+                cmdbuf.writestring(p);
+                cmdbuf.writeByte('\"');
+            }
+            if (auto p = getenv("DXSDK_DIR"))
+            {
+                // support for old DX SDK installations
+                cmdbuf.writestring(" /LIBPATH:\"");
+                cmdbuf.writestring(p);
+                cmdbuf.writestring(x64 ? `\Lib\x64"` : `\Lib\x86"`);
+            }
+            return cmdbuf.extractString();
+        }
+
+        const(char)* linkerPath(bool x64)
+        {
+            if (auto p = getVCDir(VCDir.Bin, false)) // prefer 32-bit linker in case of cross-compilation
+            {
+                OutBuffer cmdbuf;
+                cmdbuf.writestring(p);
+                cmdbuf.writestring(r"\link.exe");
+                if (VSInstallDir)
+                {
+                    // debug info needs DLLs Common7\IDE for most linker versions
+                    const char* path = getenv("PATH");
+                    const char* npath = FileName.combine(VSInstallDir, r"Common7\IDE");
```

Comment:
`path` and `npath` appear to never be used?

Reply:
Ooops, missing PATH modification.